crates/swc_ecma_parser/src/parser/expr.rs

```diff
@@ -841,7 +844,6 @@ impl<I: Tokens> Parser<I> {
         let return_type = if !self.ctx().will_expect_colon_for_cond
             && self.input.syntax().typescript()
             && is!(self, ':')
-            && !self.ctx().dont_parse_colon_as_type_ann
```

Comment:
I think we don't need `will_expect_colon_for_cond`, because `will_expect_colon_for_cond` is used to check should we parse `:` as type or not, also I found we duplicate logic, i.e. set `will_expect_colon_for_cond` and `dont_parse_colon_as_type_ann` to the same value